tabs/configuration.js

```diff
@@ -436,6 +451,104 @@ TABS.configuration.initialize = function (callback, scrollPosition) {
         $('input[name="currentoffset"]').val(BF_CONFIG.currentoffset);
         $('input[name="multiwiicurrentoutput"]').prop('checked', MISC.multiwiicurrentoutput);
 
+        var escProtocols = {
+            0: {
+                name: "STANDARD",
+                defaultRate: 400,
+                rates: {
+                    50: "50Hz",
+                    400: "400Hz"
+                }
+            },
+            1: {
+                name: "ONESHOT125",
+                defaultRate: 1000,
+                rates: {
+                    400: "400Hz",
+                    1000: "1kHz",
+                    2000: "2kHz",
+                    4000: "4kHz"
+                }
+            },
+            2: {
+                name: "ONESHOT42",
+                defaultRate: 2000,
+                rates: {
+                    400: "400Hz",
+                    1000: "1kHz",
+                    2000: "2kHz",
+                    4000: "4kHz",
+                    8000: "8kHz"
+                }
+            },
+            3: {
+                name: "MULTISHOT",
+                defaultRate: 2000,
+                rates: {
+                    400: "400Hz",
+                    1000: "1kHz",
+                    2000: "2kHz",
+                    4000: "4kHz",
+                    8000: "8kHz"
+                }
+            },
+            4: {
+                name: "BRUSHED",
+                defaultRate: 8000,
+                rates: {
+                    500: "500Hz",
+                    1000: "1kHz",
+                    2000: "2kHz",
+                    4000: "4kHz",
+                    8000: "8kHz",
+                    16000: "16kHz",
+                    32000: "32kHz"
+                }
+            }
+        };
+
+        function buildMotorRates() {
+            var protocolData = escProtocols[ADVANCED_CONFIG.motorPwmProtocol];
+
+            $escRate.find('option').remove();
+
+            for (var i in protocolData.rates) {
+                if (protocolData.rates.hasOwnProperty(i)) {
+                    $escRate.append('<option value="' + i + '">' + protocolData.rates[i] + '</option>');
+                }
+            }
+
+        }
+
+        if (semver.gte(CONFIG.flightControllerVersion, "1.3.0")) {
+
+            var $escProtocol = $('#esc-protocol');
+            var $escRate = $('#esc-rate');
+            for (var i in escProtocols) {
+                if (escProtocols.hasOwnProperty(i)) {
+                    var protocolData = escProtocols[i];
+                    $escProtocol.append('<option value="' + i + '">' + protocolData.name + '</option>');
+                }
+            }
+
+            buildMotorRates();
+            $escProtocol.val(ADVANCED_CONFIG.motorPwmProtocol);
+            $escRate.val(ADVANCED_CONFIG.motorPwmRate);
```

Comment:
What happens when FC returns something that's not on the list? Value will be set to default?